### PR TITLE
BuildKite support

### PR DESCRIPTION
**Pull request by mboes**
Migrate Nixpkgs Linux job, which is flaky on CircleCI due to
underpowered resource classes, to BuildKite.

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+steps:
+  - command: |
+      . /var/lib/buildkite-agent/.nix-profile/etc/profile.d/nix.sh
```
Comment (flokli) [inline .buildkite/pipeline.yml]: This is very specific to how our agent is set up - I'd rather see us add a `nix=true` tag to all nix-enabled agents (which set this locally in an environment hook), and then [require that tag to be present for that specific step](https://buildkite.com/docs/agent/v3/cli-start#agent-targeting).

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      nix-shell --arg docTools false --pure --run \
+          'bazel coverage //tests/... --config ci --build_tag_filters "coverage-compatible" --test_tag_filters "coverage-compatible" --test_output=all'
+    label: "Run tests (Nixpkgs)"
+    timeout: 30
+  - command: |
+      echo "common:ci --build_tag_filters -requires_hackage,-requires_lz4,-requires_zlib,-requires_doctest,-requires_c2hs,-requires_threaded_rts,-dont_test_with_bindist" > .bazelrc.local
+      # XXX: See .bazelrc [backward compatible options] for the the rational behind this flag
+      echo "build --incompatible_use_python_toolchains=false" >> .bazelrc.local
+      bazel build --config ci //tests/...
+      ./tests/run-start-script.sh
```
Comment (flokli) [inline .buildkite/pipeline.yml]: While tinkering with sourcing the nix profile inside the agent hooks, I noticed this step cannot run when the nix profile is sourced before, due to `ld.gold` not finding `gmp`. Is this a known issue / expected?

https://buildkite.com/tweag-1/rules-haskell/builds/29#b7b15f5e-178c-4bc4-b891-6e2bc2332cb7/21-174

File: .buildkite/pipeline.yml (hunk 1 above)
Comment (mboes) [reply .buildkite/pipeline.yml]: I don't see this as a problem for now: the agent such that it is setup currently is temporary. We'll want to reprovision it as completely immutable infrastructure (not in scope for this PR).

File: .buildkite/pipeline.yml (hunk 2 above)
Comment (mboes) [reply .buildkite/pipeline.yml]: Yes, I noticed this too. I'm not sure why this is. But it seems fair to keep Nix and non-nix environments separate.

Comment (flokli) [reply .buildkite/pipeline.yml]: Can you open an issue about this, so we can follow-up on it later?

Comment (flokli) [reply .buildkite/pipeline.yml]: so there's already https://github.com/tweag/rules_haskell/issues/704 about that, which we closed as WONTFIX. We might want to revisit that, assuming the CI agents run on NixOS, and we want to test the bindist feature.

**Comment by flokli** [comment]
@mboes this would need a force-merge, as CircleCI fails (because we remove its pipeline configuration)

**Pull request merged**